### PR TITLE
Adds AoT compilation support / Angular 2.1.1 update

### PR DESCRIPTION
**Pull request by Ninja101**
Added AoT compilation
Updated Angular to 2.1.1

Tested with my project, it compiles fine when using the output from 'gulp compileAot'.

### CODE REVIEW

**Comment by Ninja101** [comment]
#21

**Comment by pleerock** [comment]
thanks for PR

File: src/Modal.ts [hunk 1]
```diff
@@ -1,15 +1,39 @@
...
     template: `
 <div class="modal" 
      tabindex="-1"
      role="dialog"
      #modalRoot
      (keydown.esc)="closeOnEscape ? close() : 0"
-     [ngClass]="{ in: isOpened, fade: isOpened }"
-     [ngStyle]="{ display: isOpened ? 'block' : 'none' }"
+     [class]="{ in: isOpened, fade: isOpened }"
+     [style]="{ display: isOpened ? 'block' : 'none' }"
```
Comment (pleerock) [inline src/Modal.ts]: can you please explain these changes

**Comment by Ninja101** [comment]
@pleerock 

The compiler raised errors with ngClass/ngStyle, swapped them out and it was fine.

And as for Object.assign, the compiler defaulted it's type to ObjectConstructor which doesn't have .assign

**Comment by Ninja101** [comment]
Will create a new pull request when properly fixed

**Comment by pleerock** [comment]
would be great, thanks

**Pull request closed without merging**